### PR TITLE
[fix] debian 8 boost asio ipv6 address missing std::array 16 constructor

### PR DESCRIPTION
**Pull request by rfree-d**

### CODE REVIEW

File: src/utils/boost_asio_helper.cpp [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+#include "utils/boost_asio_helper.hpp"
+
+boost::asio::ip::address_v6 make_address( std::array<unsigned char,16> thebytes) {
```
Comment (robertoleksy) [inline src/utils/boost_asio_helper.cpp]: double copy of array

File: src/utils/boost_asio_helper.hpp [hunk 2]
```diff
@@ -0,0 +1,11 @@
+
+#pragma once
+
+#include <array>
+#include <boost/asio.hpp>
+
+/**
+  * Make address from std::array, as this conversion would not work on older versions od Boost (e.g. Debian 8 default)
```
Comment (mikurys) [inline src/utils/boost_asio_helper.hpp]: typo "od"

Comment (mikurys) [inline src/utils/boost_asio_helper.hpp]: And also not working with new version of boost

File: src/utils/boost_asio_helper.cpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	return addr;
+}
+
+} // namespace detail
+
+
+boost::asio::ip::address_v6 make_ipv6_address(const std::array<unsigned char,16> & thebytes) {
+	return detail::make_ipv6_address_conversion<
+			boost::asio::ip::address_v6 , // address format we want
+			boost::asio::ip::address_v6::bytes_type // byte format thatwe provide
```
Comment (robertoleksy) [inline src/utils/boost_asio_helper.cpp]: typo `thatwe`

**Comment by robertoleksy** [comment]
remove mo files

File: src/utils/boost_asio_helper.cpp [hunk 4]
```diff
@@ -0,0 +1,26 @@
+
+#include "utils/boost_asio_helper.hpp"
+#include <tuple>
```
Comment (mikurys) [inline src/utils/boost_asio_helper.cpp]: not needed include

Comment (rfree-d) [inline src/utils/boost_asio_helper.cpp]: 🆗

File: src/utils/misc.hpp [hunk 5]
```diff
@@ -1,4 +1,3 @@
-
```
Comment (mikurys) [inline src/utils/misc.hpp]: ?

File: src/test/boost_asio_helper.cpp [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+#include "../utils/boost_asio_helper.hpp"
+#include <libs0.hpp>
+
+TEST(boost_asio_helper, make_ipv6_address_from_bytes) {
+	boost::asio::ip::address_v6 addr1{ boost::asio::ip::address_v6::from_string("fd42:9288:db10:f1a8:29ee:28bf:17bf:a630") };
+
+	auto addr2bytes = std::array<unsigned char,16>{ { 0xfd,0x42, 0x92,0x88, 0xdb,0x10, 0xf1,0xa8,
+	0x29,0xee, 0x28,0xbf, 0x17,0xbf, 0xa6,0x30 } };
+
+	addr1.from_string("fd42:9288:db10:f1a8:29ee:28bf:17bf:a630");
```
Comment (mikurys) [inline src/test/boost_asio_helper.cpp]: why?

**Pull request merged**